src/Propel/Runtime/Adapter/Pdo/MysqlAdapter.php

```diff
@@ -131,6 +132,41 @@ public function applyLimit(&$sql, $offset, $limit)
         }
     }
 
+    /**
+     * @param Criteria $criteria
+     *
+     * @return string
+     */
+    public function getGroupBy(Criteria $criteria)
```

Comment:
How might we reduce the logic nesting in this method?

The first opportunity I see is moving the groupBy into early return.

Could aliases be removed from the $selected part of the foreach in an efficient way?